### PR TITLE
feat(hcu): enable LightOp paged_mqa_logits_fp4 on DCU

### PR DESCRIPTION
**Pull request by Rich-Bao**
- add SGLANG_USE_LIGHTOP_PAGED_MQA_LOGITS_FP4 switch, default off
- fallback to Triton kernel when lightop unavailable
- add lightop_fp4_indexer.md doc and unit test

---
### CI States

Latest PR Test (Base): :x: **Missing `run-ci` label** -- add it to run CI tests.
Latest PR Test (Extra): :x: **Blocked** -- `run-ci` is required first.

### CODE REVIEW

File: python/sglang/kernels/ops/attention/dsv4/fp4_indexer.py [hunk 1]
```diff
@@ -505,15 +547,25 @@ def fp4_index_logits_decode(
...
+            # second SGLang allowlist or swallow native execution errors.
+            return native(q, weights, slots, lens, table, page_size, out)
     BLOCK_L = 64
     grid = (B, triton.cdiv(L, BLOCK_L))
     _fp4_index_logits_kernel[grid](
         q,
         weights,
         slots,
-        lens.to(torch.int64).contiguous(),
+        lens,
```
Comment (RyeYuan) [inline python/sglang/kernels/ops/attention/dsv4/fp4_indexer.py]: 确认是general bug吗，不是的话，我们不走这里，不要改动原有代码

File: python/sglang/kernels/ops/attention/dsv4/fp4_indexer.py [hunk 2]
```diff
@@ -505,15 +547,25 @@ def fp4_index_logits_decode(
     weights = weights.to(torch.bfloat16).contiguous()
     slots = slots.contiguous()
     out = torch.empty((B, L), dtype=torch.float32, device=q.device)
-    if L == 0:
+    if B == 0 or L == 0:
         return out
+    if lens.dtype != torch.int64 or not lens.is_contiguous():
```
Comment (RyeYuan) [inline python/sglang/kernels/ops/attention/dsv4/fp4_indexer.py]: 这里加_is_hcu进行highlight标定

File: python/sglang/srt/environ.py [hunk 3]
```diff
@@ -1013,6 +1013,9 @@ class Envs:
     SGLANG_ROCM_USE_AITER_TILELANG_MHC = EnvBool(False)
     SGLANG_USE_DPSKV4_LIGHTOP_RMSNORM = EnvBool(False)
     SGLANG_USE_DPSKV4_LIGHTOP_QUANT_K_CACHE = EnvBool(False)
+    # Opt in to LightOp FP32 logits for the DSV4 FP4 decode/verify indexer on
+    # gfx928/936/938. Missing kernels or unsupported layouts retain Triton.
+    SGLANG_USE_LIGHTOP_PAGED_MQA_LOGITS_FP4 = EnvBool(False)
```
Comment (RyeYuan) [inline python/sglang/srt/environ.py]: 确认是否对所有并行方式性能优化生效，如果是，默认改成True，减少CLI上对环境变量的依赖
